### DISCUSSION TITLE
Diagnostics of IExecutable/IQueryable before execution

### DISCUSSION THREAD

**Discussion by jbockle** [ChilliCream/hotchocolate#5578] [Q&A]
Having a filterable collection field with a resolver that returns IQueryable<T>/IExecutable<T>, shouldn't I be able to create an `ExecutionDiagnosticEventListener` so I can add a breakpoint/`Print()`/`ToString()` it for troubleshooting before it is executed?  I don't know which method or which property on the context that would have the iexecutable with any queryable operations staged on it.

**Comment by PascalSenn** [comment]
This is not possible with an `ExecutionDiagnosticEventListener` as you are either to early or to late in the resolver pipeline. You either get nothing or the already executed result.

What you can do is you can add a middleware that prints this out:
```csharp
public class Query
{
    [Report]
    [UseFiltering]
    public IExecutable<Customer> GetCustomers() => new[]
    {
        new Customer()
        {
            Id = 1
        },
        new Customer()
        {
            Id = 2
        }
    }.AsExecutable();
}

public class ReportAttribute : ObjectFieldDescriptorAttribute
{
    protected override void OnConfigure(
        IDescriptorContext context, IObjectFieldDescriptor descriptor, MemberInfo member)
    {
        descriptor.Use(next => async context =>
        {
            await next(context);
            if (context.Result is IExecutable executable)
            {
                Console.WriteLine(executable.Print());
            }
        });
    }
}
```
